src/ibtracert.c

```diff
@@ -811,9 +820,29 @@ int main(int argc, char **argv)
 	argc -= optind;
 	argv += optind;
 
-	if (argc < 2)
+	if (argc < 2 && lid_file[0] == '\0') 
 		ibdiag_show_usage();
 
+	if (argc == 2) {
+		srcid = strdup(argv[0]);
+		dstid = strdup(argv[1]);
+     	}
+
+	if (lid_file != NULL) {
+		lid_fd = fopen(lid_file, "r");
+		if (!lid_fd) {
+			IBEXIT("cannot open lid-file %s", lid_file);
+		} else {
```

Comment:
No need for the else here...  you are exiting if the file does not get opened.

Reply:
Ah - was not familiar with restructured text and therefore did not see the rst directory.  I made the changes directly in the .in file - thinking it was the "input" to the actual man page.  Will go back and do it right. ( as clearly documented in README.rst :)
Will also redo the code bit that includes the unnecessary else.
You can refuse (or whatever) this one and I'll submit a new one.

The use case is grabbing a "route map" for a large number of nodes - perhaps a job allocation or compute-to-IO, etc.  Calling ibtracert every time is expensive.  As a test, I created a route map for 128 nodes with this patch in just under 3 minutes.  Doing the same thing  by calling ibtracert individually each time took over 21 minutes.

You can see now why Howard was hoping for a friendly maintainer.

Reply:
Ok.

Could you include this use case in the commit message for the series.

It would also be interesting to evaluate if ibtracert is issuing duplicate queries to the switch tables for each of the S/DLID pairs.  If so that would be another optimization.  [3 minutes still seems like a long time...]

Reply:
It depends on topology of network and number of pairs whose path is being traced. How many hops end to end ?